lib/puppet/parser/resource.rb

```diff
@@ -184,6 +184,15 @@ def name
   # if we ever receive a parameter named 'tag', set
   # the resource tags with its value.
   def set_parameter(param, value = nil)
+    if evaluated?
+      strict = Puppet[:strict]
+      if :off != strict
```

Comment:
I really dislike this style, especially with a negation - would much rather read `unless strict == :off` than `if strict != :off`, and than `if :off != strict`. I know this is for the reason of picking Symbol's equate, and possibly for performance, but it is horrible to read.